CDCgov/prime-reportstream
# Fixed empty dependency array in useEffect

This PR removes the empty dependency array for `useEffect` in the `<FacilitiesTable>` component. To fix this, I've added the `getFacilities` function and `facilitiesURL` variable as dependencies to `useEffect`. This resulted in infinite calls to the API due to the facilities function being "remade" endlessly. This is why you'll now find `getFacilities` wrapped in the `useCallback` hook with a dependency that does not change. This means the function will ONLY be recreated if that dependency changes, thus `useEffect` is not calling this function endlessly.

Test Steps:
1. pull `kevinhaube/react-build-errors`
2. `npm run build`
3. Ensure the following warning isn't present: `React Hook useEffect has a missing dependency: 'getFacilities'. Either include it or remove the dependency array`
4. `npm start`
5. Press F12 and ensure the facilities call is not repeated infinitely manytimes

## Changes
- Removed empty dependency array from `useEffect`

## Checklist

### Testing

### Process

## Fixes
- #2377 

## To Be Done
*Create GitHub issues to track the work remaining, if any*
- #issue

## Specific Security-related subjects a reviewer should pay specific attention to
- Does this PR introduce new endpoints?
No
- Does this PR include changes in authentication and/or authorization of existing endpoints?
No
- Does this change introduce new dependencies that need vetting?
No
- Does this change require changes to our infrastructure?
No
- Does logging contain sensitive data?
No
- Does this PR include or remove any sensitive information itself?
No

If you answered '_yes_' to any of the questions above, conduct a detailed Review that addresses at least:

- What are the potential security threats and mitigations? Please list the _STRIDE_ threats and how they are mitigated
    - **S**poofing (faking authenticity)
        - Threat _T_, which could be achieved by _A_, is mitigated by _M_
    - **T**ampering (influence or sabotage the integrity of information, data, or system)
    - **R**epudiation (the ability to dispute the origin or originator of an action)
    - **I**nformation disclosure (data made available to entities who should not have it)
    - **D**enial of service (make a resource unavailable)
    - **E**levation of Privilege (reduce restrictions that apply or gain privileges one should not have)
- Have you ensured logging does not contain sensitive data?
- Have you received any additional approvals needed for this change?
